org.opent2t.sample.multisensor.superpopular/com.wink.sensorpod/js/thingTranslator.js

```diff
@@ -70,10 +70,15 @@ class StateReader {
 }
 
 function convertDeviceDateToTranslatorDate(unixTimestamp) {
-    // Date takes a number of milliseconds, so convert seconds to milliseconds
-    var datetime = new Date(unixTimestamp * 1000);
+    // Only convert if input is valid numeric value
+    if (unixTimestamp && !isNaN(unixTimestamp)) { 
+        // Date takes a number of milliseconds, so convert seconds to milliseconds
+        var datetime = new Date(unixTimestamp * 1000);
+        return datetime.toISOString();
+    }
 
-    return datetime.toISOString();
+    // return all other invalid input as-is since we can't convert to DateTime
+    return unixTimestamp;
```

Comment:
I see - thanks Ben esp because i am ramping up on this sensor type -- but our schema is likely derived off the WINK schema yes? i see your point however looks like updated_at and changed_at might not always be the same - and if WINK distinguishes shouldnt we too? Isnt it going to get very confusing if we start to cook up values that might not jive with what the user might see if they were to use a real WINK app?  But maybe i dont quite get the scenario yet  and what we are aiming for with this translation in the first place. If we (in opent2t) are flexible about conflating updatedAt/ChangedAt then sure i like the idea of falling back to udpated, but wouldnt that mean another query/lookup to the server for the property ? I havent yet investigated if we query for the entire object in one go or if use targeted queries;   Also I'd much rather go for simpler logic here but then again i am not sure what our goal is. I'll let the others esp Arjun weigh in also; I dont mind the default epoch time for the default case but it again might get confusing and misleading. 

I think this needs a broader discussion here -- this change at least unblocks hte package publish; What i think the next step could be to also add a telemetry event here to try to see what are the different cases where we fail to convert and re-evaluate our behavior/schema ?

---
In reply to: [106049051](https://github.com/openT2T/translators/pull/210#discussion_r106049051) [](ancestors = 106049051)